test/math.jl

```diff
@@ -760,6 +760,102 @@ end
     @test quadgk(cos, 0,0.7,1, norm=abs)[1] ≈ sin(1)
 end
 
+module Test19626
+    using Base.Test
+    # Begin tests for 19626: Unitful compatibility.
+    # Define a mock physical quantity type
+    immutable MockQuantity{T} <: Number
+        val::T
+    end
+    Base.promote_rule{T,S}(::Type{MockQuantity{T}}, ::Type{MockQuantity{S}}) =
+        MockQuantity{promote_type(T,S)}
+    Base.convert{T}(::Type{MockQuantity{T}}, x::MockQuantity) = MockQuantity(T(x.val))
+
```

Comment:
You shouldn't need to define promotion rules and conversion for a minimal working example.   Just define the minimum number of methods necessary for a test.